Fixes invalid assert

boudewijn-tribler opened the pull request:
See #155.

boudewijn-tribler committed:
Fixes invalid assert.

boudewijn-tribler committed:
Assert cleanup.

NielsZeilemaker commented on database.py:
```diff
@@ -361,7 +361,8 @@ def executemany(self, statement, sequenceofbindings):
             assert isinstance(sequenceofbindings, (tuple, list, set)), "The sequenceofbindings must be a tuple, list, or set"
             assert all(isinstance(x, (tuple, list, dict, set)) for x in list(sequenceofbindings)), "The sequenceofbindings must be a list with tuples, lists, dictionaries, or sets"
-            assert not filter(lambda x: filter(lambda y: isinstance(y, str), x), list(sequenceofbindings)), "The bindings may not contain a string. \nProvide unicode for TEXT and buffer(...) for BLOB."
+            assert all(all(not isinstance(y, str) for y in x) for x in sequenceofbindings), "The bindings may not contain a string. \nProvide unicode for TEXT and buffer(...) for BLOB."
```
This assert now causes a dict to have non-string keys for namedparameter queries.

boudewijn-tribler committed:
Less verbose logging.

boudewijn-tribler committed:
Fixes old bug in assert.

Using dictionaries as bindings would test the dictionary keys instead of
the dictionary values.

NielsZeilemaker commented on database.py:
```diff
@@ -340,9 +347,9 @@ def executemany(self, statement, sequenceofbindings):
-        @type bindings: list containing tuples
+        @param sequenceofbindings: a list, tuple or set of bindings, where every binding contains
+        the values that must be set to the placeholders in statement.
+        @type bindings: list, tuple or set
```
or dict

boudewijn-tribler replied:
the sequenceofbindings can not be a dict, although the sequence can contain dictionaries.

the `@type bindings` should be `@type sequenceofbindings`, as there is no parameter called bindings.  And sequenceofbindings can also be a generator.

boudewijn-tribler committed:
Updated docstring.

boudewijn-tribler commented:
segfaults.  Elric did a system upgrade, lets see if it helps.

boudewijn-tribler commented:
retest this please
